pkg/report/diffreport.go

```diff
@@ -0,0 +1,22 @@
+package report
+
+import (
+	"github.com/Azure/gocover/pkg/gittool"
+	"golang.org/x/tools/cover"
+)
+
+// NewDiffCoverageReport creates a diff coverage report instance.
+// TODO: implement it
+func NewDiffCoverageReport() DiffCoverageReport {
+	return nil
+}
+
+// TODO: implement DiffCoverageReport interface
+type DiffCoverageReport interface {
+	// DiffCoverage calculate diff coverage
+	DiffCoverage(profile []*cover.Profile, path gittool.Patch) error
+	// ApplyFilter filters the cover profile with filters
+	ApplyFilter(profile []*cover.Profile, filters []string) error
```

Comment:
curious why ApplyFilter is a standalone function.
it may be more proper to make this a parameter of construction of DiffCoverageReport?

Reply:
Yeah. It's better to make `ApplyFilter` as an internal operation for DiffCoverageReport. Expose it only want to show that there are some filters operations.